custom/icds/data_management/models.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime
+
+from django.db import models
+from django.utils.translation import ugettext_noop
+
+from custom.icds.data_management.const import DATA_MANAGEMENT_TASKS
+
+
+class DataManagementRequest(models.Model):
+    STATUS_PENDING = 0
+    STATUS_IN_PROGRESS = 1
+    STATUS_SUCCESS = 2
+    STATUS_FAILED = 3
+    STATUS_CHOICES = (
+        (STATUS_PENDING, ugettext_noop('pending')),
+        (STATUS_IN_PROGRESS, ugettext_noop('in progress')),
+        (STATUS_SUCCESS, ugettext_noop('successful')),
+        (STATUS_FAILED, ugettext_noop('failed'))
+    )
+    STATUSES = [STATUS_PENDING, STATUS_IN_PROGRESS, STATUS_SUCCESS, STATUS_FAILED]
+    slug = models.CharField(max_length=255, blank=False, null=False)
+    domain = models.CharField(max_length=255, blank=False, null=False)
+    db_alias = models.CharField(max_length=255, blank=True, null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    initiated_by = models.CharField(max_length=255, blank=False, null=False)
+
+    # timestamps of request runtime
+    started_on = models.DateTimeField()
+    ended_on = models.DateTimeField(blank=True, null=True)
+
+    # to consider cases modified within range
+    from_date = models.DateField(blank=True, null=True)
+    till_date = models.DateField(blank=True, null=True)
+
+    error = models.TextField(blank=True, null=True)
+    status = models.SmallIntegerField(choices=STATUS_CHOICES, default=STATUS_PENDING)
+
+    def execute(self):
+        """
+        run the updates through DataManagement
+        """
+        self._set_in_progress()
+        if self.slug not in DATA_MANAGEMENT_TASKS:
+            self._note_error("Unexpected slug %s" % self.slug)
+        else:
+            try:
+                self._perform_task()
+            except Exception as e:
+                self._note_error(str(e))
+                raise
+            else:
+                self.status = self.STATUS_SUCCESS
+            finally:
+                self.ended_on = datetime.utcnow()
+                self.save()
+
+    def _set_in_progress(self):
+        self.status = self.STATUS_IN_PROGRESS
+        self.started_on = datetime.utcnow()
+        self.save()
+        self.refresh_from_db()
```

Comment:
why is this necessary?

Reply:
We are updating the record itself, so to refresh it? Even though it might not appear to be useful, but just wanted to get any changes that might happen on save like `created_at` would get set on this save.

Reply:
I'm almost 100% sure that any values updated in the DB will be set on the model during the save call. I think this is only needed if you expect there to be changes directly in the DB or else from a separate instance of the model.

Reply:
okay cool. Just was thinking of a request instance being rerun and modified in parallel in another process, an edge case not worth refetching record for.
https://github.com/dimagi/commcare-hq/pull/26610/commits/0802b61dc7c6c1398e02f9baef600f338bc4a7ae